vulture_os/toolkit/api_parser/carbon_black/carbon_black.py

```diff
@@ -147,32 +152,37 @@ def format_log(self, log):
         return json.dumps(log)
 
     def execute(self):
-
         since = self.last_api_call or (datetime.utcnow() - timedelta(hours=24))
+        to = datetime.now().strftime("%Y-%m-%dT%H:%M:%S.000Z")
 
-        available = 0
-        found = 0
-        while available <= found:
-
-            response = self.get_logs(available, since)
+        cursor = 1
+        while True:
+            response = self.get_logs(cursor, since, to)
 
             # Downloading may take some while, so refresh token in Redis
             self.update_lock()
 
             logs = response['results']
-            found = int(response['num_found'])
+            found = len(logs)
+            available = int(response.get('num_available', "0"))
+            # logger.debug(f"cursor: {cursor}, found: {found}, available: {available}", extra={'frontend': str(self.frontend)})
             # available key is not present if response = {'results': [], 'num_found': 0}
-            available = int(response.get('num_available', "0")) + 1
 
-            self.write_to_file([self.format_log(l) for l in logs])
+            if found == 0:
+                break
+
+            self.write_to_file([self.format_log(log) for log in logs])
 
             # Writting may take some while, so refresh token in Redis
             self.update_lock()
 
-            if len(logs) > 0:
-                # API returns alerts sorted, first is newer
-                # Replace "Z" by "+00:00" for datetime parsing
-                self.frontend.last_api_call = datetime.fromisoformat(logs[0]['last_update_time'].replace("Z", "+00:00"))+timedelta(milliseconds=1)
+            # API returns alerts sorted, first is newer
+            # Replace "Z" by "+00:00" for datetime parsing
+            self.frontend.last_api_call = datetime.fromisoformat(
+                logs[-1]['backend_update_timestamp'].replace("Z", "+00:00")) + timedelta(milliseconds=1)
+            cursor += found
 
-        logger.info(f"[{__parser__}]:execute: Parsing done.", extra={'frontend': str(self.frontend)})
+            if cursor > available:
+                break
```

Comment:
What if there are more than 10k logs available? the API doesn't handle more, that may fail if the query is not adapted to fetch less logs